### ISSUE TITLE
pmon crashing immediately in latest master image

### ISSUE THREAD

**Issue by daall** [Azure/sonic-buildimage#5759]
**Description**
pmon is consistently crashing in the latest master image. It looks like some import/package is missing, causing it to fail suddenly on startup.

**Steps to reproduce the issue:**
1. Load the latest master image onto a device and reboot

**Describe the results you received:**
```
Oct 30 17:10:58.161675 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:55,590 CRIT Set uid to user 0
Oct 30 17:10:58.161751 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:55,590 INFO Included extra file "/etc/supervisor/conf.d/platform_api_server.conf" during parsing
Oct 30 17:10:58.161914 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:55,590 INFO Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
Oct 30 17:10:58.161980 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:55,619 INFO RPC interface 'supervisor' initialized
Oct 30 17:10:58.162071 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:55,619 CRIT Server 'unix_http_server' running without any HTTP authentication checking
Oct 30 17:10:58.162129 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:55,620 INFO supervisord started with pid 1
Oct 30 17:10:58.162297 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:56,623 INFO spawned: 'dependent-startup' with pid 14
Oct 30 17:10:58.162365 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:56,627 INFO spawned: 'supervisor-proc-exit-listener' with pid 15
Oct 30 17:10:58.162455 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:56,632 INFO spawned: 'platform_api_server' with pid 16
Oct 30 17:10:58.162527 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:56,785 INFO exited: platform_api_server (exit status 1; not expected)
Oct 30 17:10:58.162688 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:58,091 INFO success: dependent-startup entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Oct 30 17:10:58.162752 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:58,092 INFO success: supervisor-proc-exit-listener entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Oct 30 17:10:58.162842 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:58,094 INFO spawned: 'platform_api_server' with pid 20
Oct 30 17:10:58.162902 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:58,133 INFO spawned: 'rsyslogd' with pid 21
Oct 30 17:10:58.213904 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server Traceback (most recent call last):
Oct 30 17:10:58.214048 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server   File "/opt/platform_api_server.py", line 10, in <module>
Oct 30 17:10:58.214111 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server     import sonic_platform
Oct 30 17:10:58.214165 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server ImportError: No module named sonic_platform
Oct 30 17:11:00.817377 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server Traceback (most recent call last):
Oct 30 17:11:00.818110 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server   File "/opt/platform_api_server.py", line 10, in <module>
Oct 30 17:11:00.818657 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server     import sonic_platform
Oct 30 17:11:00.819120 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server ImportError: No module named sonic_platform
Oct 30 17:11:01.322180 str2-dx010-acs-6 INFO pmon#supervisord: start sonic-platform package not installed, attempting to install...
Oct 30 17:11:02.149704 str2-dx010-acs-6 INFO pmon#supervisord: start ERROR: sonic_platform-1.0-py2-none-any.whl is not a supported wheel on this platform.
Oct 30 17:11:04.261641 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server Traceback (most recent call last):
Oct 30 17:11:04.261641 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server   File "/opt/platform_api_server.py", line 10, in <module>
Oct 30 17:11:04.261641 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server     import sonic_platform
Oct 30 17:11:04.261641 str2-dx010-acs-6 INFO pmon#supervisord: platform_api_server ImportError: No module named sonic_platform
Oct 30 17:11:04.331792 str2-dx010-acs-6 INFO systemd[1]: run-docker-runtime\x2drunc-moby-de1a943afd274d6b6dac562fc8c7d0d090771b4ab2f1e9bac735888ddf4d31b4-runc.by00qm.mount: Succeeded.
Oct 30 17:11:07.838617 str2-dx010-acs-6 INFO systemd[1]: run-docker-runtime\x2drunc-moby-f137af5982cd06d539b4fba918059eac35b4af41d1ebf80dad8de83dd3021be8-runc.2Ty1H3.mount: Succeeded.
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:58,224 INFO exited: platform_api_server (exit status 1; not expected)
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:10:59,707 INFO success: rsyslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:00,711 INFO spawned: 'platform_api_server' with pid 25
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:00,788 INFO spawned: 'start' with pid 26
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:00,790 INFO success: start entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:00,830 INFO exited: platform_api_server (exit status 1; not expected)
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:04,153 INFO spawned: 'platform_api_server' with pid 34
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:04,271 INFO exited: platform_api_server (exit status 1; not expected)
Oct 30 17:11:08.169741 str2-dx010-acs-6 INFO pmon#supervisord 2020-10-30 17:11:04,272 INFO gave up: platform_api_server entered FATAL state, too many start retries too quickly
```

```
admin@sonic:~$ docker ps -a
CONTAINER ID        IMAGE                                COMMAND                  CREATED             STATUS                     PORTS               NAMES
fc6e07faa791        docker-sonic-telemetry:latest        "/usr/bin/supervisord"   10 hours ago        Up 5 minutes                                   telemetry
356fae419d6e        docker-sonic-mgmt-framework:latest   "/usr/bin/supervisord"   10 hours ago        Up 2 hours                                     mgmt-framework
89ee098418cc        docker-snmp:latest                   "/usr/bin/supervisord"   10 hours ago        Up 4 minutes                                   snmp
c83f6e2c069f        docker-router-advertiser:latest      "/usr/bin/docker-iniâ€¦"   10 hours ago        Up 4 minutes                                   radv
b369b07fdc93        docker-dhcp-relay:latest             "/usr/bin/docker_iniâ€¦"   10 hours ago        Up 4 minutes                                   dhcp_relay
4b71e3b9dd2c        docker-lldp:latest                   "/usr/bin/docker-lldâ€¦"   10 hours ago        Up 5 minutes                                   lldp
de1a943afd27        docker-syncd-brcm:latest             "/usr/bin/supervisord"   10 hours ago        Up 4 minutes                                   syncd
812464d5669a        docker-teamd:latest                  "/usr/bin/supervisord"   10 hours ago        Up 4 minutes                                   teamd
af3360d98aed        docker-orchagent:latest              "/usr/bin/docker-iniâ€¦"   10 hours ago        Up 4 minutes                                   swss
e74a45cb4e78        docker-fpm-frr:latest                "/usr/bin/docker_iniâ€¦"   10 hours ago        Up 5 minutes                                   bgp
54d621332b0f        docker-platform-monitor:latest       "/usr/bin/docker_iniâ€¦"   10 hours ago        Exited (0) 2 minutes ago                       pmon
f137af5982cd        docker-database:latest               "/usr/local/bin/dockâ€¦"   10 hours ago        Up 2 hours                                     database
```

**Describe the results you expected:**
pmon should work.

**Output of `show version`:**
```
admin@sonic:~$ show ver

SONiC Software Version: SONiC.master.461-6088bd59
Distribution: Debian 10.6
Kernel: 4.19.0-9-2-amd64
Build commit: 6088bd59
Build date: Thu Oct 29 19:06:46 UTC 2020
Built by: sonic@jenkins-worker-8

Platform: x86_64-cel_seastone-r0
HwSKU: Celestica-DX010-C32
ASIC: broadcom
Serial Number: DX010F2B118711MS100005
Uptime: 17:21:48 up  2:03,  1 user,  load average: 9.85, 6.60, 5.43

Docker images:
REPOSITORY                    TAG                   IMAGE ID            SIZE
docker-snmp                   latest                da467be983be        426MB
docker-snmp                   master.461-6088bd59   da467be983be        426MB
docker-teamd                  latest                29df3c93ee45        425MB
docker-teamd                  master.461-6088bd59   29df3c93ee45        425MB
docker-sonic-mgmt-framework   latest                5c067f4b99f2        549MB
docker-sonic-mgmt-framework   master.461-6088bd59   5c067f4b99f2        549MB
docker-router-advertiser      latest                071d37bcd803        390MB
docker-router-advertiser      master.461-6088bd59   071d37bcd803        390MB
docker-platform-monitor       latest                a74214f9bd0f        503MB
docker-platform-monitor       master.461-6088bd59   a74214f9bd0f        503MB
docker-lldp                   latest                03cb08ffa877        456MB
docker-lldp                   master.461-6088bd59   03cb08ffa877        456MB
docker-dhcp-relay             latest                57b818251555        397MB
docker-dhcp-relay             master.461-6088bd59   57b818251555        397MB
docker-database               latest                92fb2157bf9d        390MB
docker-database               master.461-6088bd59   92fb2157bf9d        390MB
docker-orchagent              latest                462cc6634084        437MB
docker-orchagent              master.461-6088bd59   462cc6634084        437MB
docker-nat                    latest                524f117bfe7c        428MB
docker-nat                    master.461-6088bd59   524f117bfe7c        428MB
docker-sonic-telemetry        latest                7235ee54ccbd        459MB
docker-sonic-telemetry        master.461-6088bd59   7235ee54ccbd        459MB
docker-fpm-frr                latest                736695efe884        441MB
docker-fpm-frr                master.461-6088bd59   736695efe884        441MB
docker-sflow                  latest                8e9c073d8380        428MB
docker-sflow                  master.461-6088bd59   8e9c073d8380        428MB
docker-syncd-brcm             latest                e9af209ac842        527MB
docker-syncd-brcm             master.461-6088bd59   e9af209ac842        527MB
```

[syslog.txt](https://github.com/Azure/sonic-buildimage/files/5466644/syslog.txt)

**Comment by msosyak** [comment]
I also faced a similar issue with pmon on the latest SONiC image. The error is a bit different, but the issue is observed after #5743 and #5747 was merged, See details below.
@jleveque  Could you check or should I open a new issue?

```
Nov  2 17:38:23.591661 igk-dut-05 WARNING pmon#psud[34]: Failed to load chassis due to ImportError('No module named enum - required module not found',)
Nov  2 17:38:23.610616 igk-dut-05 ERR pmon#ledd[31]: Failed to load ledutil: No module named enum - required module not found
Nov  2 17:38:23.613121 igk-dut-05 INFO pmon#supervisord: ledd Failed to load ledutil: No module named enum - required module not found
Nov  2 17:38:23.624429 igk-dut-05 ERR pmon#psud[34]: Failed to load psuutil: No module named enum - required module not found
Nov  2 17:38:23.626334 igk-dut-05 INFO pmon#supervisord: psud Failed to load psuutil: No module named enum - required module not found
Nov  2 17:38:23.643568 igk-dut-05 WARNING pmon#xcvrd[33]: Failed to load chassis due to ImportError('No module named enum - required module not found',)
Nov  2 17:38:23.654974 igk-dut-05 ERR pmon#xcvrd[33]: Failed to load sfputil: No module named enum - required module not found
Nov  2 17:38:23.658881 igk-dut-05 INFO pmon#supervisord: xcvrd Failed to load sfputil: No module named enum - required module not found
Nov  2 17:38:23.662182 igk-dut-05 INFO pmon#supervisord: thermalctld Traceback (most recent call last):
Nov  2 17:38:23.662362 igk-dut-05 INFO pmon#supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 711, in <module>
Nov  2 17:38:23.662517 igk-dut-05 INFO pmon#supervisord: thermalctld     main()
Nov  2 17:38:23.662657 igk-dut-05 INFO pmon#supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 707, in main
Nov  2 17:38:23.662872 igk-dut-05 INFO pmon#supervisord: thermalctld     thermal_control.run()
Nov  2 17:38:23.663010 igk-dut-05 INFO pmon#supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 654, in run
Nov  2 17:38:23.663156 igk-dut-05 INFO pmon#supervisord: thermalctld     import sonic_platform.platform
Nov  2 17:38:23.663294 igk-dut-05 INFO pmon#supervisord: thermalctld   File "/usr/lib/python2.7/dist-packages/sonic_platform/__init__.py", line 8, in <module>
Nov  2 17:38:23.663437 igk-dut-05 INFO pmon#supervisord: thermalctld     from . import (
Nov  2 17:38:23.663588 igk-dut-05 INFO pmon#supervisord: thermalctld   File "/usr/lib/python2.7/dist-packages/arista/utils/sonic_platform/platform.py", line 11, in <module>
Nov  2 17:38:23.663732 igk-dut-05 INFO pmon#supervisord: thermalctld     raise ImportError("%s - required module not found" % e)
Nov  2 17:38:23.663864 igk-dut-05 INFO pmon#supervisord: thermalctld ImportError: No module named enum - required module not found
Nov  2 17:38:23.667497 igk-dut-05 WARNING pmon#syseepromd[35]: Failed to load platform-specific eeprom from sonic_platform package due to ImportError('No module named enum - required module not found',)


admin@igk-dut-05:~$ show version 

SONiC Software Version: SONiC.master.25-dirty-20201102.081530
Distribution: Debian 10.6
Kernel: 4.19.0-9-2-amd64
Build commit: f2a258ac
Build date: Mon Nov  2 08:31:15 UTC 2020
Built by: johnar@jenkins-worker-4

Platform: x86_64-arista_7170_64c
HwSKU: Arista-7170-64C
ASIC: barefoot
Serial Number: SGD19412172
Uptime: 17:30:04 up  2:09,  1 user,  load average: 1.76, 1.82, 1.54
```

**Comment by jleveque** [comment]
@msosyak: I also discovered this issue on Friday; it applies to Arista devices. The Arista sonic_platform package imports 'enum', which is part of the standard library starting with Python 3.4. However, for versions < 3.4, the 'enum34' package needs to be installed. This package was previously getting installed in PMon for Python 2.7, but due to conflicts between these two packages in Python 3.7, I recently made a change to eliminate the need for the 'enum' package by all daemons in the container. However, I was unaware that the Arista sonic_platform package still had a dependency. I believe I found a way to prevent conflicts between these packages, and have opened a PR to bring back the dependency [here](https://github.com/Azure/sonic-platform-daemons/pull/107), which should ensure that the 'enum34' package is installed for Python 2.7 but not for Python 3.7. This **should** also fix the Arista issue, as the package will be installed in the container when the xcvrd package is installed.

However, this type of problem could arise in the future, because the Arista sonic_platform package is not installed in the PMon container like it is done for all other vendors. Instead, the package installation directory is mounted in the container from the host OS. Since the package is not installed in the container, we have to explicitly make sure to install any dependent packages in the container (with my fix above, things should luckily work because enum34 will get installed when xcvrd gets installed -- however, if this dependency is removed, like we just saw, it will break Arista platforms).

Moving forward, it would be great if the Arista sonic_platform package gets installed in the container using the same mechanism we use for other vendors, as then the wheel package could properly specify all its dependencies, ensuring they get installed at the same time. @Staphylo for awareness.

**Comment by daall** [comment]
@jleveque LGTM in the latest master image. I'm going to go ahead and close this issue. Do we need to open another issue for the Arista issue, or does the PR you opened fully cover it?

**Comment by jleveque** [comment]
@daall: The two PRs I mention above should fix the issue. I say you can close this.

@msosyak: If, after those PRs merge, the Arista issue still exists, please open a new issue to track it.

**Comment by daall** [comment]
Cool, thanks again! :)
